File: Busqueda renglones-columnas 03-09-25/recorrido_matriz.cpp
#include <iostream>
using namespace std;

int main() {
    int matriz[3][3] = {
        {1, 2, 3},
        {4, 5, 6},
        {7, 8, 9}
    };

    cout << "Matriz original:" << endl;
    for (int i = 0; i < 3; i++) {
        cout << "[ ";
        for (int j = 0; j < 3; j++) {
            cout << matriz[i][j] << " ";
        }
        cout << "]" << endl;
    }

    cout << "\n En filas:" << endl;
    for (int i = 0; i < 3; i++) {
        for (int j = 0; j < 3; j++) {
            cout << matriz[i][j] << " ";
        }
    }

    cout << "\n En columnas:" << endl;
    for (int j = 0; j < 3; j++) {
        for (int i = 0; i < 3; i++) {
            cout << matriz[i][j] << " ";
        }
    }

    return 0;
}

File: Busqueda renglones-columnas 03-09-25/recorrido_matriz.js
let matriz = [
    [1, 2, 3],
    [4, 5, 6],
    [7, 8, 9]
];

console.log("Matriz original:");
for (let i = 0; i < matriz.length; i++) {
    console.log("[ " + matriz[i].join(" ") + " ]");
}

console.log("\nElementos en filas:");
for (let i = 0; i < matriz.length; i++) {
    for (let j = 0; j < matriz[i].length; j++) {
        process.stdout.write(matriz[i][j] + " ");
    }
}

console.log("\nElementos en columnas:");
for (let j = 0; j < matriz[0].length; j++) {
    for (let i = 0; i < matriz.length; i++) {
        process.stdout.write(matriz[i][j] + " ");
    }
}
